src/models/Trace.js

```diff
@@ -0,0 +1,28 @@
+import { PUCK_POSITION } from '../const/puck.const';
+import { CHARGE_SIZE } from '../const/charge.const';
+
+export default class Trace {
+  constructor() {
+    this.previousPosition = [{ x: null, y: null }];
```

Comment:
Initialize as an empty array, there is no need to include object containing structure / null values